src/OpenCensus.Exporter.ApplicationInsights/Implementation/TraceExporterHandler.cs

```diff
@@ -82,5 +203,233 @@ public void Export(IList<ISpanData> spanDataList)
                 this.telemetryClient.Track(result);
             }
         }
+
+        private void ExtractGenericProperties(ISpanData span, out SpanKind resultKind, out DateTimeOffset timestamp, out string name, out string resultCode, out IDictionary<string, string> props, out string traceId, out string spanId, out string parentId, out Tracestate tracestate, out bool? success, out TimeSpan duration)
+        {
+            resultKind = span.Kind;
+
+            // TODO: Should this be a part of generic logic?
+            if (resultKind == SpanKind.Unspecified)
+            {
+                if (span.HasRemoteParent.HasValue && span.HasRemoteParent.Value)
+                {
+                    resultKind = SpanKind.Server;
+                }
+                else
+                {
+                    resultKind = SpanKind.Client;
+                }
+            }
+
+            // 1 tick is 100 ns
+            timestamp = DateTimeOffset.FromUnixTimeSeconds(span.StartTimestamp.Seconds);
+            timestamp = timestamp.Add(TimeSpan.FromTicks(span.StartTimestamp.Nanos / 100));
+
+            name = span.Name;
+
+            props = new Dictionary<string, string>();
```

Comment:
could you avoid creating a new dictionary and copying properties? instead just pass telemetry properties here by ref?

Reply:
It's harder here. I will think how to restructure code later alongside other perf improvements